configs/configs.go

```diff
@@ -4,321 +4,227 @@ package configs
-		if os.Getenv("GHORG_BITBUCKET_USERNAME") == "" {
-			return ErrNoBitbucketUsername
-		}
-	}
-
-	if len(token) != tokenLength {
-		if scmProvider == "github" {
+// VerifyToken checks to make sure env is set for the correct scm provider
+func (config *Config) VerifyToken() error {
+	switch config.ScmType {
+	case "github":
+		if len(config.Token) != 40 {
 			return ErrNoGitHubToken
 		}
-
-		if scmProvider == "gitlab" {
+	case "gitlab":
+		if len(config.Token) != 20 {
 			return ErrNoGitLabToken
 		}
-
-		if scmProvider == "bitbucket" {
+	case "bitbucket":
+		if len(config.Token) != 20 {
 			return ErrNoBitbucketAppPassword
 		}
+		if config.BitbucketUsername == "" {
+			return ErrNoBitbucketUsername
+		}
+	}
+	return nil
+}
 
+// VerifyClone makes sure flags are set to appropriate values
+func (config *Config) VerifyClone() error {
+	if config.CloneType != "user" && config.CloneType != "org" {
+		return ErrIncorrectCloneType
+	}
+
+	if config.CloneProtocol != "ssh" && config.CloneProtocol != "https" {
+		return ErrIncorrectProtocolType
 	}
 
 	return nil
 }
 
-// VerifyConfigsSetCorrectly makes sure flags are set to appropriate values
-func VerifyConfigsSetCorrectly() error {
-	scmType := os.Getenv("GHORG_SCM_TYPE")
-	cloneType := os.Getenv("GHORG_CLONE_TYPE")
-	protocol := os.Getenv("GHORG_CLONE_PROTOCOL")
-
-	if !utils.IsStringInSlice(scmType, scm.SupportedClients()) {
-		return ErrIncorrectScmType
+// Print shows the user what is set before cloning
+func (config *Config) Print() {
+	colorlog.PrintInfo("*************************************")
+	colorlog.PrintInfo("* SCM           : " + config.ScmType)
+	colorlog.PrintInfo("* Type          : " + config.CloneType)
+	colorlog.PrintInfo("* Protocol      : " + config.CloneProtocol)
+	colorlog.PrintInfo("* Location      : " + config.Path)
+	colorlog.PrintInfo(fmt.Sprintf("* Concurrency   : %d", config.Concurrency))
+	colorlog.PrintInfo("* Branch        : " + config.getGhorgBranch())
+	colorlog.PrintInfo("* Base URL      : " + config.ScmBaseUrl)
+	colorlog.PrintInfo(fmt.Sprintf("* Skip Archived : %t", config.SkipArchived))
+	colorlog.PrintInfo(fmt.Sprintf("* Skip Forks    : %t", config.SkipForks))
+	colorlog.PrintInfo(fmt.Sprintf("* Backup        : %t", config.Backup))
+	if ghorgIgnoreDetected() {
+		colorlog.PrintInfo("* Ghorgignore   : true")
 	}
+	colorlog.PrintInfo("* Output Dir    : " + config.Path)
 
-	if cloneType != "user" && cloneType != "org" {
-		return ErrIncorrectCloneType
+	colorlog.PrintInfo("*************************************")
+	fmt.Println("")
+}
+
+func (config *Config) getGhorgBranch() string {
+	if config.Branch == "" {
+		return "default branch"
 	}
 
-	if protocol != "ssh" && protocol != "https" {
-		return ErrIncorrectProtocolType
+	return config.Branch
+}
+
+func (config *Config) parseOutputDir(argz []string) {
+	if config.Path != "" {
+		return
 	}
 
-	return nil
+	config.Path = HomeDir() + "/Desktop/ghorg/" + strings.ToLower(strings.ReplaceAll(argz[0], "-", "_"))
```

Comment:
@Code0x58 thanks for the comments and congrats on the new job! 

> windows portability issues due to paths

The HomeDir library supports [windows](https://github.com/mitchellh/go-homedir/blob/master/homedir.go#L41) so I'd always assumed it would work on windows, however I don't have a windows machine and have never tested, nor have I gotten any issues about issues running on windows. I'm guessing the path.filepath.Join function performs something similar to the HomeDir package, but I don't know much about windows environments.

> setting a default target branch name (e.g. master) will not work in some cases

What do you mean by this? The default branch is obtained from the repo itself, which should generally always be set, if not it will fall back to master, which I think is a safe bet. Otherwise we would need to skip the clone, I think it would be better to try to clone something than not and if we're going to try something master is a good bet. If we make the user choose a branch to clone they will miss a large portion of the repos. Mostly because github has recently changed the default branch from master to main. 

> defaulting to ~/ghorg/{repo name} 

This makes sense. Originally when I started ghorg it cloned the org to the desktop. I wanted to append _ghorg so the users knew it was a ghorg folder and to avoid collisions. Now that its under the ghorg directory it doesn't make as much sense. 

> why transform
 
I like the the transformation because it standardizes all the directories in the ghorg folder, just bugged me when they were not all uniform.

Reply:
> @Code0x58 thanks for the comments and congrats on the new job!

:D Thanks!

> windows portability issues due to paths

Windows uses `\` as a path separator instead of `/` so you can use `Join` to simply generate system specific paths. I took it for granted as a practice over a decade ago (e.g python's `os.path.join`) but after checking now, it is not required by Windows' APIs itself but may be relied on by some tools (e.g. CLI) - maybe it's not needed, and maybe even bad to do without a specific case. I think you're right about HomeDir supporting windows and getting the right directory.

A bit of a tangent: you can add binary releases which also cover windows without much trouble by updating the GitHub Actions, e.g. [workflow](https://github.com/Code0x58/gitlab-ci-validate/blob/master/.github/workflows/release.yaml) and [release](https://github.com/Code0x58/gitlab-ci-validate/releases/tag/v0.3.3).

> setting a default target branch name (e.g. master) will not work in some cases

I seem to recall it looked like there's some places were either explicitly setting `config.Branch` to a literal like master, or conditional branches which would use a literal like master if something else isn't explicitly set, rather than reading it from the HEAD symbolic-ref of the remote or host API. I've found that to be a case I've seen a fair bit while cloning wild repos (e.g. [drive to move from master to main](https://github.com/github/renaming)).